fcrepo-exts/fcrepo-camel-toolbox
# Upgrades Fixity Service

**Upgrades Fixity service to Camel 3.9.0.**
* * *

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3760

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
Brings the fixity service in line with other camel toolbox microservices by
1) removing osgi dependencies
2) upgrade to Camel 3.9
3) fixity service can be enabled using the `fixity.enabled=true` property.


# What's new?

# How should this be tested?
in your fcrepo-camel-toolbox.properties file:
```
fixity.enabled=true
fixity.input.stream=broker:queue:fixity
fixity.success=file:/tmp/?fileName=fixity-success.log&fileExist=Append
```

Start fedora: 

```
docker run --rm -p8080:8080 -p61616:61616  -p8181:8181 --name=my_fcrepo6  fcrepo/fcrepo:latest
```

Add a binary resource
```
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/fcrepo/rest/hello-world  -H "Content-Type: text/plain"  -X PUT --data "hello world"
```

start camel toolbox:

```
java -jar fcrepo-camel-toolbox-app-6.0.0-SNAPSHOT-driver.jar -c fcrepo-camel-toolbox.properties
```

Perform fixity check
```
 curl -XPOST localhost:9080/reindexing -H"Content-Type: application/json" \
        -d '["broker:queue:fixity"]'
```

Verify the results are found in the success file

```
cat /tmp/fixity-success.log
```

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers
